### PR TITLE
Encapsulate batch writer logic into WriteBuffer

### PR DESCRIPTION
**Pull request by mkalinin**
### What was done?
- Move batch writer logic to `WriteBuffer`
- Remove standalone `BatchWriter` class

### CODE REVIEW

**Review by ericsson49 — APPROVED** [review]
LGTM

File: db/core/src/main/java/org/ethereum/beacon/db/source/WriteBuffer.java [hunk 1]
```diff
@@ -88,12 +88,12 @@ public void remove(@Nonnull final K key) {
...
-      for (Map.Entry<K, CacheEntry<V>> entry : buffer.entrySet()) {
-        if (entry.getValue() == CacheEntry.REMOVED) {
-          getUpstream().remove(entry.getKey());
-        } else {
-          getUpstream().put(entry.getKey(), entry.getValue().value);
-        }
+      if (getUpstream() instanceof BatchUpdateDataSource) {
+        final Map<K, V> updates = new HashMap<>();
+        buffer.forEach((key, value) -> updates.put(key, value.value));
+        ((BatchUpdateDataSource<K, V>) getUpstream()).batchUpdate(updates);
```
Comment (Nashatyrev) [inline db/core/src/main/java/org/ethereum/beacon/db/source/WriteBuffer.java:94]: What about `REMOVED` entries?

Comment (mkalinin) [reply db/core/src/main/java/org/ethereum/beacon/db/source/WriteBuffer.java:94]: That's a good question. `REMOVED` cache entry has a value equal to `null` which preserves a null-removed contract. I'll add a comment to `batchUpdate` and `REOMVED` entry about that.

**Review by Nashatyrev — APPROVED** [review]

File: db/core/src/main/java/org/ethereum/beacon/db/source/WriteBuffer.java (hunk 1 above)
Comment (mkalinin) [reply db/core/src/main/java/org/ethereum/beacon/db/source/WriteBuffer.java:94]: Fixed https://github.com/harmony-dev/beacon-chain-java/pull/175/commits/c65f5c05995cd44871ffd322de67fb0471953586.

**Review by eustimenko — APPROVED** [review]

**Pull request merged**